src/compiler/checker.ts

```diff
@@ -11866,7 +11889,11 @@ module ts {
                 var identifier = <Identifier>name;
                 if (contextNode && (contextNode.parserContextFlags & ParserContextFlags.StrictMode) && isEvalOrArgumentsIdentifier(identifier)) {
                     var nameText = declarationNameToString(identifier);
-                    return grammarErrorOnNode(identifier, Diagnostics.Invalid_use_of_0_in_strict_mode, nameText);
+
+                    // Always report 'eval' and 'arguments' invalid usage in strict mode code regardless of parser diagnostics

```

Comment:
Thanks for catching this!.  This is from my previous implementation in which I report a parsing error for using reserve keyword so I still want to report an error in this case. @CyrusNajmabadi corrects me about reserve word. So this change is no longer necessary